core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -817,16 +816,40 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     logger.info(s"Refreshed index searchers to make reads consistent for dataset=$ref shard=$shardNum")
   }
 
-  //scalastyle:off method.length
+  val regexChars = Array('.', '?', '+', '*', '|', '{', '}', '[', ']', '(', ')', '"', '\\')
+  val regexCharsMinusPipe = (regexChars.toSet - '|').toArray
+
+  // scalastyle:off method.length
   private def leafFilter(column: String, filter: Filter): Query = {
+
+    def equalsQuery(value: String): Query = {
+      if (value.nonEmpty) new TermQuery(new Term(column, value))
+      else leafFilter(column, NotEqualsRegex(".+")) // value="" means the label is absent or has an empty value.
+    }
+
     filter match {
       case EqualsRegex(value) =>
         val regex = removeRegexAnchors(value.toString)
-        if (regex.replaceAll("\\.\\*", "") == "")
```

Comment:
Previously, `""` would get caught by this `if` condition and `MatchAllDocsQuery` would be used. Now, `NotEqualsRegex(".+")` is used. Is that intentional?

Also, I don't think this is correct:
```
          // if label=~"" then match empty string or label not present condition too
           leafFilter(column, NotEqualsRegex(".+"))
```
-- since that regex would not match any empty strings.

Reply:
@amolnayak311  's PRs earlier had some regressions. This PR reorders the ifs and restores behavior for `label=~""` to behave prior to his change. So this begs the question. Was @yu-shipit the right behavior ?

Lets discuss in person.

Reply:
~"" is converted to NotEqualsRegex(".+") to handle non-existing label. 
EqualsRegex(".*")  requires that the label exists.

Reply:
I've tested `label=~""` in a Prometheus server, and it looks like the behavior there matches this `NotEqualsRegex(".+")` filter.